python/cudf/cudf/bindings/sort.pyx

```diff
@@ -182,7 +182,10 @@ class SegmentedRadixSortPlan(object):
 
         # Note: .astype is required below because .copy_to_device
         #       is just a plain memcpy
-        d_begins.copy_to_device(cudautils.astype(segments, dtype=seg_dtype))
+        import cudf.bindings.typecast as typecast
+        from cudf.dataframe import columnops
+        col = typecast.apply_cast(columnops.as_column(segments), dtype=seg_dtype)
+        d_begins.copy_to_device(col.data.mem)
```

Comment:
This is really awkward, col is newly allocated typecast and then we immediately copy the result into `d_begins`.

Reply:
Talked offline about this and this code is confusing and will be ripped out soon anyway, so it's fine for now.